### PR TITLE
Revise javadoc for sdk, state, options, annotations, window, (and misc)

### PR DESCRIPTION
**Pull request by kennknowles**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`.
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [x] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

R: @tgroh perhaps, though any/all input is helpful! Feel free to hot potato.

CC: @davorbonaci this should certainly be cherry picked

### CODE REVIEW

File: sdks/java/core/src/main/java/org/apache/beam/sdk/Pipeline.java [hunk 1]
```diff
@@ -329,9 +313,14 @@ public CoderRegistry getCoderRegistry() {
...
 
+  /////////////////////////////////////////////////////////////////////////////
+  // Below here are operations that aren't normally called by users.
+
   /**
-   * Sets the {@link CoderRegistry} that this {@link Pipeline} uses.
+   * @deprecated this should never be used - every {@link Pipeline} has a registry throughout its
+   *     lifetime.
    */
+  @Deprecated
```
Comment (tgroh) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/Pipeline.java]: `Internal`?

File: sdks/java/core/src/main/java/org/apache/beam/sdk/PipelineRunner.java [hunk 2]
```diff
@@ -25,15 +25,14 @@
...
+ * A {@link PipelineRunner} runs a {@link Pipeline}.
  *
- * @param <ResultT> the type of the result of {@link #run}.
+ * @param <ResultT> the type of the result of {@link #run}, often a handle to a running job.
  */
 public abstract class PipelineRunner<ResultT extends PipelineResult> {
 
   /**
-   * Constructs a runner from the provided options.
+   * Constructs a runner from the provided {@link PipelineOptions}.
```
Comment (tgroh) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/PipelineRunner.java]: Perhaps it is worth adding an additional note that subclasses of `PipelineRunner` must declare this method as well?

File: sdks/java/core/src/main/java/org/apache/beam/sdk/Pipeline.java [hunk 3]
```diff
@@ -284,19 +274,16 @@ public void visitPrimitiveTransform(Node node) {
...
    */
   public PipelineResult run() {
     return run(defaultOptions);
   }
 
   /**
-   * Runs this {@link Pipeline} using the given {@link PipelineOptions}, using the runner
-   * specified by the options.
+   * Runs this {@link Pipeline} using the given {@link PipelineOptions}, using the runner specified
+   * by the options.
```
Comment (tgroh) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/Pipeline.java]: Note that this method should generally be assumed to be asynchronous, and reference `PipelineResult#waitUntilFinish()`

File: sdks/java/core/src/main/java/org/apache/beam/sdk/annotations/Experimental.java [hunk 4]
```diff
@@ -24,25 +24,20 @@
...
 
 /**
- * Signifies that a public API (public class, method or field) is subject to
- * incompatible changes, or even removal, in a future release. An API bearing
- * this annotation is exempt from any compatibility guarantees made by its
- * containing library. Note that the presence of this annotation implies nothing
- * about the quality or performance of the API in question, only the fact that
- * it is not "API-frozen."
+ * Signifies that a public API (public class, method or field) is subject to incompatible changes,
+ * or even removal, in a future release.
```
Comment (tgroh) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/annotations/Experimental.java]: "or even removal" -> "including removal"?

File: sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/AfterAll.java [hunk 5]
```diff
@@ -28,7 +28,8 @@
 import org.joda.time.Instant;
 
 /**
- * Create a {@link Trigger} that fires and finishes once after all of its sub-triggers have fired.
+ * A composite {@link Trigger} that fires and finishes once after all of its sub-triggers have
```
Comment (tgroh) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/AfterAll.java]: rm "and finishes", or add it to the other `OnceTriggers`

File: sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/GlobalWindows.java [hunk 6]
```diff
@@ -23,7 +23,7 @@
 import org.joda.time.Instant;
 
 /**
- * Default {@link WindowFn} that assigns all data to the same window.
+ * A default {@link WindowFn} that assigns all data to the same window.
```
Comment (tgroh) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/GlobalWindows.java]: rm 'default', or "The default"

File: sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/MergeOverlappingIntervalWindows.java [hunk 7]
```diff
@@ -23,7 +23,9 @@
 import java.util.List;
 
 /**
- * A utility function for merging overlapping {@link IntervalWindow}s.
+ * <b>For internal use only; no backwards compatibility guarantees.</b>
+ *
+ * <p>A utility function for merging overlapping {@link IntervalWindow}s.
```
Comment (tgroh) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/MergeOverlappingIntervalWindows.java]: `@Internal`?

"for merging" -> "which merges"

File: sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/Repeatedly.java [hunk 8]
```diff
@@ -22,7 +22,7 @@
 import org.joda.time.Instant;
 
 /**
- * Repeat a trigger, either until some condition is met or forever.
+ * A composite {@link Trigger} that fires according to its subtrigger forever.
```
Comment (tgroh) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/Repeatedly.java]: do we need "compoiste" here?

File: sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/Sessions.java [hunk 9]
```diff
@@ -25,16 +25,17 @@
 import org.joda.time.Duration;
 
 /**
- * A {@link WindowFn} windowing values into sessions separated by {@link #gapDuration}-long
- * periods with no elements.
+ * A {@link WindowFn} windowing values into sessions separated by periods with no input for at least
```
Comment (tgroh) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/Sessions.java]: "A WindowFn that windows values..."

File: sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/WindowMappingFn.java [hunk 10]
```diff
@@ -19,15 +19,20 @@
...
 import org.joda.time.Duration;
 
 /**
+ * <b>For internal use only; no backwards compatibility guarantees.</b>
+ *
  * A function that takes the windows of elements in a main input and maps them to the appropriate
  * window in a {@link PCollectionView} consumed as a
  * {@link MultiOutput#withSideInputs(PCollectionView[]) side input}.
  */
+@Internal
```
Comment (tgroh) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/WindowMappingFn.java]: We expect this to be (eventually) user-accessible, yes? `Experimental` seems more appropriate.

File: sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/AfterAll.java (hunk 5 above)
Comment (kennknowles) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/AfterAll.java]: Done (via rephrase)

File: sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/GlobalWindows.java (hunk 6 above)
Comment (kennknowles) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/GlobalWindows.java]: Done (via rephrase)

File: sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/MergeOverlappingIntervalWindows.java (hunk 7 above)
Comment (kennknowles) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/MergeOverlappingIntervalWindows.java]: Done

File: sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/Repeatedly.java (hunk 8 above)
Comment (kennknowles) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/Repeatedly.java]: Removed.

File: sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/Sessions.java (hunk 9 above)
Comment (kennknowles) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/Sessions.java]: Done

File: sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/WindowMappingFn.java (hunk 10 above)
Comment (kennknowles) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/WindowMappingFn.java]: Done.

We do expect users to use it eventually, but probably a long time from now and very rarely.

**Pull request merged**